include/convex_bodies/ellipsoidintersectconvex.h

```diff
@@ -0,0 +1,284 @@
+// VolEsti (volume computation and sampling library)
+
+// Copyright (c) 2012-2023 Vissarion Fisikopoulos
+// Copyright (c) 2018-2023 Apostolos Chalkis
+// Copyright (c) 2029-2023 Elias Tsigaridas
+
+// Licensed under GNU LGPL.3, see LICENCE file
+
+#ifndef ELLIPSOIDINTERSECTCONVEX_H
+#define ELLIPSOIDINTERSECTCONVEX_H
+
+/// This class represents a polytope intersected with an ellipsoid
+/// \tparam Polytope Polytope Type
+/// \tparam CEllipsoid Ellipsoid Type
+template <typename Polytope, typename CEllipsoid>
+class EllipsoidIntersectPolytope {
```

Comment:
Why not inherit from BallIntersectPolytope class? A lot of methods are the same. Also do we need this class at all or we can just use BallIntersectPolytope for both ball and ellipsoid intersections? The only difference I see is in the last 3 methods.